aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -203,5 +212,50 @@ Tensor& logical_xor_(Tensor& self, const Tensor& other) {
   return native::logical_xor_out(self, self, other);
 }
 
+Tensor& lt_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(result.scalar_type() == kBool,
+      "The output tensor of lt must be a bool, but was ", result.scalar_type());
+  auto iter = TensorIterator::comparison_op(result, self, other,
+      /*check_mem_overlap=*/true);
+  lt_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor lt(const Tensor& self, const Tensor& other) {
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  native::lt_out(result, self, other);
+  return result;
+}
+
+Tensor& lt_(Tensor& self, const Tensor& other) {
+  TORCH_CHECK(self.dtype() == other.dtype(),
+      "Expected object of scalar type ", self.dtype(), " but got scalar type ",
+      other.dtype(), " for argument 'other' in call to lt_")
+  auto iter = TensorIterator::comparison_op(self, self, other,
+      /*check_mem_overlap=*/true);
+  lt_stub(iter.device_type(), iter);
+  return self;
+}
+
+// First it will validate that is possible to convert scalar to tensor dtype without overflow.
+// It's different for arithmetic ops. For arithmetic ops we don't verify this.
+// The reason is because previous TH behavior checked for overflow in comparison ops but not arithmetic ops.
+// In the future, we should reconsider this inconsistency and decide if we're going to check overflow for arithmetic ops
```

Comment:
Sorry, when I wrote all of this I was in a hurry. Let's make it more concise:
```
// `lt` validates that is possible to convert Scalar other to self's dtype without overflow.
// This behavior is unique to comparison ops; arithmetic operations don't do this.
// In the future, we should reconsider this inconsistency and decide if we want to add the same check to arithmetic ops.
```

Reply:
Thanks